traits/trait_list_object.py

```diff
@@ -416,6 +416,21 @@ def rename(self, name):
             self.name = name
             self.trait = trait.handler
 
+    def clear(self):
+        if self.trait.minlen > 0:
+            self.len_error(0)
+
+        if len(self) == 0:
```

Comment:
Can we restructure slightly so that `list.clear` is still called when `len(self) == 0`? Right now `list.clear` on an empty list is almost certainly a no-op, but we can't know for sure that Python won't add some functionality at some point in the future.